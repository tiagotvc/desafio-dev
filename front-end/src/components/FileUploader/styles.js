import styled from 'styled-components';

export const ContainerUpload = styled.div`
    display:flex;
    align-items: center ;
    justify-content:center ;

    .input_container {
        display:flex ;
        align-items:center ;
        justify-content:center ;
    }

    .upload_preview_container {
        padding:20px;
        display: flex ;
    }

    .drop-file-input {
        position: relative;
        width: 280px;
        height: 140px;
        border: 2px dashed black;
        border-radius: 20px;
        display: flex;
        align-items: center;
        justify-content: center;   
    }

    .drop-file-input input {
        opacity: 0;
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        cursor: pointer;
    }

    .drop-file-input:hover,
    .drop-file-input.dragover {
        opacity: 0.6;
    }

    .drop-file-input__label {
        text-align: center;
        color: gray;
        font-weight: 600;
        padding: 10px;
    }

    .drop-file-input__label img {
        width: 100px;
    }

    
`


export const ContainerPreview = styled.div`
    display:flex;
    align-items: center ;
    justify-content:center ;display:block;

    .drop-file-preview {
        margin-top: 10px;
        display: block ;
    }

    .drop-file-preview__title {
        display:flex;
        align-items: center ;
        justify-content:center ;
    }

    .drop-file-preview_items {
        display:flex ;
        justify-content:center ;
        align-items:center ;
    }

    .drop-file-preview p {
        font-weight: 100;
        font-size: 13px ;
    }

    .drop-file-preview__item {
        display:flex ;
        align-items: center ;
        justify-content: center ;
        
}

    .drop-file-preview__item img {
        height:40px ;
        
    }

    .drop-file-preview__item__info {
      justify-content:center ;
      align-items:center ;
      text-align:center ;
     
    }

    .drop-file-preview__item__del {
        width: 40px;
        height: 40px;
        border-radius: 50%;
        display: flex;
        margin-left:350px;
        position: absolute;
        cursor: pointer;
        
    }

    .drop-file-preview__item:hover .drop-file-preview__item__del {
        opacity: 1;
    }
    `