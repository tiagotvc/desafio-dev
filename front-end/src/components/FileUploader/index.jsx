import React from 'react';
import * as Styled from './styles';

export const FileUploader = (props) => {

    return (
        <></>
    )
}