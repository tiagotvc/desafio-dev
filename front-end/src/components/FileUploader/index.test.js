import React from 'react';
import FileUploader from './index';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('FileUploader ', () => {
  it('renders correctly', () => {
    const tree = renderer
    .create(<FileUploader/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
    });
});