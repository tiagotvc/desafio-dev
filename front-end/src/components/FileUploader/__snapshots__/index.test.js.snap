// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploader  renders correctly 1`] = `
.c0 {
  display: block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .align_divs_center {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .align_divs_center > span {
  margin-top: 20px;
}

.c0 .align_divs_center > button {
  margin-top: 40px;
}

.c0 .input_container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .upload_preview_container {
  padding: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .drop-file-input {
  position: relative;
  width: 280px;
  height: 140px;
  border: 2px dashed black;
  border-radius: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .drop-file-input input {
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.c0 .drop-file-input:hover,
.c0 .drop-file-input.dragover {
  opacity: 0.6;
}

.c0 .drop-file-input__label {
  text-align: center;
  color: gray;
  font-weight: 600;
  padding: 10px;
}

.c0 .drop-file-input__label img {
  width: 100px;
}

<div
  className="c0"
>
  <div
    className="align_divs_center"
  >
    <div
      class="input_container"
      className="drop-file-input"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDrop={[Function]}
    >
      <div
        class="drop-file-input__label"
      >
        <img
          alt=""
          src="cloud-upload-regular-240.png"
        />
        <p>
          Drag & Drop your files here
        </p>
      </div>
      <input
        onChange={[Function]}
        type="file"
        value=""
      />
    </div>
  </div>
</div>
`;
